### PR TITLE
ORC-1474: replaced deprecated `getMinimum/Maximum` in TestColumnStatistics

### PR DESCRIPTION
**Pull request by mystic-lama**
### What changes were proposed in this pull request?
As part of ORC-661, getMinimum/getMaximum were deprecated.
TestColumnStatistics uses these deprecated methods.
We need to replace them as
getMinimum -> getMinimumLocalDate
getMaximum -> getMaximumLocalDate

### Why are the changes needed?
Updated deprecated methods to keep tests in sync with implementation and preventing failure when deprecated methods are removed from dependencies

### How was this patch tested?
The Unit test case passes

### CODE REVIEW

File: java/core/src/test/org/apache/orc/TestColumnStatistics.java [hunk 1]
```diff
@@ -28,6 +27,7 @@
 import org.apache.hadoop.hive.serde2.io.DateWritable;
 import org.apache.hadoop.hive.serde2.io.HiveDecimalWritable;
 import org.apache.hadoop.io.Text;
+import org.apache.hadoop.shaded.org.apache.commons.text.StringEscapeUtils;
```
Comment (wgtmac) [inline java/core/src/test/org/apache/orc/TestColumnStatistics.java:30]: Is this change required?

Comment (mystic-lama) [reply java/core/src/test/org/apache/orc/TestColumnStatistics.java:30]: @wgtmac StringEscapeUtils is deprecated. I am split on this change, hence left a note in the PR description above, as commons-text is not available as direct dependency in ORC, and not inclined to add it just for one class.
Open to suggestions

Comment (wgtmac) [reply java/core/src/test/org/apache/orc/TestColumnStatistics.java:30]: @dongjoon-hyun @williamhyun Any suggestion? I am not that familiar with that dependency.

Comment (dongjoon-hyun) [reply java/core/src/test/org/apache/orc/TestColumnStatistics.java:30]: `*.shaded.` is supposed to be used by the original project only. Depending on other project's shaded dependency is worse than using a deprecated dependency. I agree with @wgtmac 's concern. Let's remove this.

Comment (mystic-lama) [reply java/core/src/test/org/apache/orc/TestColumnStatistics.java:30]: @dongjoon-hyun Ack, let me remove this change

Comment (mystic-lama) [reply java/core/src/test/org/apache/orc/TestColumnStatistics.java]: Rolled back the change, and updated PR description and JIRA to reflect the same, to avoid confusion in future

Comment (mystic-lama) [reply java/core/src/test/org/apache/orc/TestColumnStatistics.java]: @dongjoon-hyun Thanks for the feedback. Updated the description. Does it look better?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @mystic-lama and @wgtmac .
